### ISSUE TITLE
[iOS] Content with DRM not playing on AirPlay

### ISSUE THREAD

**Issue by facugu1998** [TheWidlarzGroup/react-native-video#3411]
# Bug
When trying to play a Fairplay encrypted video in an Apple TV using Airplay, the content is not playing, and the screen is black. If the content is not encrypted, the video plays correctly. Also, we can play the DRM content in the app without problem, using the custom getLicense function.

Has anyone faced this problem? Thanks in advance

## Platform
Which player are you experiencing the problem on:
* iOS

## Environment info

Library version: 6.0.0
Device: iPhone 8 plus

**Comment by freeboub** [comment]
can it be linked to: https://github.com/react-native-video/react-native-video/issues/3406

**Comment by facugu1998** [comment]
I will test next week on a version before the PR we made and check if it worked before.

**Comment by facugu1998** [comment]
@freeboub Hello, I tested in on alpha-8 and didn't work. I found that the problem is that airplay requires new DRM keys so it uses ResourceLoaderDelegate. As the requests stored in loadingRequests are resolved when the phone request them, when airplay does the request for license, they have already been resolved but for the phone, so the keys are wrong.

The solution that worked for me was the following:
 1. Add a reset function in RCTResourceLoaderDelegate.swift:

```
public func reset(){
        _loadingRequests.removeAll();
        _requestingCertificate = false;
}
```

2. Call the reset function in "handleExternalPlaybackActiveChange" on RCTVideo.swift:

```
func handleExternalPlaybackActiveChange(player: AVPlayer, change: NSKeyValueObservedChange<Bool>) {
    guard let _player = _player else { return }
    _resouceLoaderDelegate?.reset();
    onVideoExternalPlaybackChange?(["isExternalPlaybackActive": NSNumber(value: _player.isExternalPlaybackActive),
                                    "target": reactTag as Any])
}
```

**Comment by dgocoder** [comment]
@facugu1998 `_loadingRequests.removeAll()` doesn't exist in that file. There is `_loadingRequest` but it does not have removeAll function

**Comment by venBharathManjunath** [comment]
Hi @freeboub @KrzysztofMoch

The solution mentioned by @facugu1998 is correct and it fixes the issue for Airplay Streaming with DRM. Any plans on including this changes ?

**Comment by freeboub** [comment]
@venBharathManjunath can you please open a pr with this change please? Looks like small changes easy to integrate .
I just have a doubt when we leave fair play playback , will the stream be restored correctly on the phone ?

**Comment by KrzysztofMoch** [comment]
@freeboub I remember that DRM with airplay started working after my DRM rework

**Comment by venBharathManjunath** [comment]
@KrzysztofMoch Does your change is included in version 6.4.5 ? Could you please share a reference to the changes

**Comment by freeboub** [comment]
> @KrzysztofMoch Does your change is included in version 6.4.5 ? Could you please share a reference to the changes

No, If I remember well it is in 6.6.0, so directly switch to 6.6.3 !
